src/WebJobs.Script/Description/DotNet/CSharp/CSharpCompilationService.cs

```diff
@@ -65,7 +66,7 @@ private static string GetFunctionSource(FunctionMetadata functionMetadata)
                 code = File.ReadAllText(functionMetadata.ScriptFile);
             }
 
-            return code ?? string.Empty;
+            return Utility.RemoveUtf8ByteOrderMark(code ?? string.Empty);
```

Comment:
The bug description mentions that it only repros when the BOM is NOT present in the source file, and here you're removing it?